module/module_test.go

```diff
@@ -493,192 +463,124 @@ func TestModularResourceWithoutReconfigure(t *testing.T) {
 
-	client := pb.NewModuleServiceClient(conn)
-	m.SetReady(true)
-	readyResp, err := client.Ready(ctx, &pb.ReadyRequest{ParentAddress: parentAddr})
-	test.That(t, err, test.ShouldBeNil)
-	test.That(t, readyResp.Ready, test.ShouldBeTrue)
+		_, err = m.AddResource(ctx, &pb.AddResourceRequest{
+			Config: mockReconfigConf, Dependencies: deps,
+		})
+		test.That(t, err, test.ShouldBeNil)
 
-	validateResp, err := m.ValidateConfig(ctx, &pb.ValidateConfigRequest{
-		Config: sumConf,
+		test.That(t, reconfigDeps1, test.ShouldResemble, deps)
+		test.That(t, reconfigConf1.Attributes, test.ShouldResemble, mockReconfigConf.Attributes)
+		test.That(t, reconfigConf1.ConvertedAttributes, test.ShouldImplement, &MockConfig{})
 	})
-	test.That(t, err, test.ShouldBeNil)
 
-	deps := validateResp.Dependencies
-	test.That(t, deps, test.ShouldResemble, []string{"rdk:component:motor/motor1"})
+	//nolint:dupl
+	t.Run("reconfigurable reconfiguration", func(t *testing.T) {
+		mockAttrs, err := protoutils.StructToStructPb(MockConfig{
+			Motors: []string{motor.Named("motor2").String()},
+		})
+		test.That(t, err, test.ShouldBeNil)
 
-	_, err = m.AddResource(ctx, &pb.AddResourceRequest{
-		Config: sumConf, Dependencies: deps,
-	})
-	test.That(t, err, test.ShouldBeNil)
+		mockReconfigConf.Attributes = mockAttrs
 
-	// Assert that mock is configured correctly after adding.
-	var depNames []string
-	for k := range mock.deps {
-		depNames = append(depNames, motor.Named(k.Name).String())
-	}
-	test.That(t, depNames, test.ShouldResemble, deps)
-	test.That(t, mock.config.Motors, test.ShouldResemble,
-		[]string{"rdk:component:motor/motor1"})
+		validateResp, err := m.ValidateConfig(ctx, &pb.ValidateConfigRequest{
+			Config: mockReconfigConf,
+		})
+		test.That(t, err, test.ShouldBeNil)
 
-	reconfigureDeps := []string{motor.Named("motor2").String()}
-	_, err = m.ReconfigureResource(ctx, &pb.ReconfigureResourceRequest{
-		Config: newSumConf, Dependencies: reconfigureDeps,
-	})
-	test.That(t, err, test.ShouldBeNil)
+		deps := validateResp.Dependencies
+		test.That(t, deps, test.ShouldResemble, []string{"rdk:component:motor/motor2"})
 
-	// Assert that mock is configured correctly after reconfiguring.
-	var reDepNames []string
-	for k := range mock.deps {
-		reDepNames = append(reDepNames, motor.Named(k.Name).String())
-	}
-	test.That(t, reDepNames, test.ShouldResemble, reconfigureDeps)
-	test.That(t, mock.config.Motors, test.ShouldResemble,
-		[]string{"rdk:component:motor/motor2"})
-	test.That(t, mock.reconfigureCallCount, test.ShouldEqual, 1)
+		_, err = m.ReconfigureResource(ctx, &pb.ReconfigureResourceRequest{
+			Config: mockReconfigConf, Dependencies: deps,
+		})
+		test.That(t, err, test.ShouldBeNil)
+
+		test.That(t, reconfigDeps2, test.ShouldResemble, deps)
+		test.That(t, reconfigConf2.Attributes, test.ShouldResemble, mockReconfigConf.Attributes)
+		test.That(t, reconfigConf2.ConvertedAttributes, test.ShouldImplement, &MockConfig{})
```

Comment:
Shouldn't this test the actual values within ConvertedAttributes? Not just that it implements MockConfig{}? Zeroed out MockConfig{} would pass this test but not be valid behavior.